cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1990,21 +1992,36 @@ availableSourceTargets :: ElaboratedConfiguredPackage
                        -> [(PackageId, ComponentName, Bool,
                             AvailableTarget (UnitId, ComponentName))]
 availableSourceTargets elab =
-    -- Now this is slightly odd: we look at all the components of the package
-    -- to which this graph node corresponds. For graph nodes that are whole
-    -- packages this makes perfect sense, but for nodes corresponding to
-    -- components this sort-of amounts to taking the cross-product of the set
-    -- of components of a package with itself. Obviously this will give rise
-    -- to duplicates. The reason we do it is because it lets us reconstruct
-    -- the "missing" components that got stripped from the plan because they
-    -- were disabled (or a couple other reasons). We need those missing
-    -- components since they are things the user could refer to and we need
-    -- to know about them and have enough info about them to explain why they
-    -- cannot be selected.
+    -- We have a somewhat awkward problem here. We need to know /all/ the
+    -- components from /all/ the packages because these are the things that
+    -- users could refer to. Unfortunately at this stage the elaborated install
+    -- plan does /not/ contain all components: some components have already
+    -- been deleted because they cannot possibly be built. This is the case
+    -- for components that are marked @buildable: False@ in their .cabal files.
+    -- (It's not unreasonable that the unbuildable components have been pruned
+    -- as the plan invariant is considerably simpler if all nodes can be built)
+    --
+    -- We can recover the missing components but it's not exactly elegant. For
+    -- a graph node corresponding to a component we still have the information
+    -- about the package that it came from, and this includes the names of
+    -- /all/ the other components in the package. So in principle this lets us
+    -- find the names of all components, plus full details of the buildable
+    -- components.
+    --
+    -- Consider for example a package with 3 exe components: foo, bar and baz
+    -- where foo and bar are buildable but baz is not. So the plan contains
+    -- nodes for the components foo and bar. Now we look at each of these two
```

Comment:
"Now, " (comma)? Or do you mean this in the sense of "we used to do X, now we do Y"?